tests/test_features_feature_collection.py

```diff
@@ -24,11 +24,51 @@
 )
 from tsflex.utils.data import flatten
 
-from .utils import dummy_data
+from .utils import dummy_data, dummy_group_data  # noqa: F401
 
 ## FeatureCollection
 
 
+def test_single_series_group_feature_collection(dummy_group_data):
+    fd = FeatureDescriptor(
+        function=np.sum,
+        series_name="count",
+    )
+
+    fc = FeatureCollection(feature_descriptors=fd)
+
+    assert fc.get_required_series() == ["count"]
+    assert fc.get_nb_output_features() == 1
+    res_list = fc.calculate(dummy_group_data, group_by="country", return_df=False)
+    res_df = fc.calculate(dummy_group_data, group_by="country", return_df=True)
+
+    assert isinstance(res_list, list)
+    assert isinstance(res_df, pd.DataFrame)
+    assert np.all(dummy_group_data["country"].unique() == res_df.index.unique())
+
+    for idx, frame in enumerate(res_list):
+        assert_frame_equal(frame, res_df.iloc[(idx) : (idx + 1)])
```

Comment:
Do we have a guarantee that the series list to df-column mapping is one-to-one? 
(I think it just will always work here because you only have 1 output feature, so there are no reshuffle possibilities)

I believe, this is a too big assumption, and you must use the `frame.name` as identifier (the column-names should be exactly the same)